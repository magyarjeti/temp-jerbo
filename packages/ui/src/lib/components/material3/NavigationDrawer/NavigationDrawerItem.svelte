<script lang="ts">
	import JetiIcon from '../../JetiIcon.svelte';
	import { type Snippet } from 'svelte';
	import { getNavigationDrawerContext } from './navigationDrawer.svelte';
	import StateLayer from '../StateLayer.svelte';
	import { concatClasses } from '@packages/toolbox/concatClasses';
	import type { HTMLAnchorAttributes, HTMLButtonAttributes } from 'svelte/elements';
	import { isAnchorAttrs, isButtonAttrs } from '../../../utils/components';

	type Props = {
		icon?: string | null;
		active?: boolean;
		children?: Snippet;
	} & (HTMLAnchorAttributes | HTMLButtonAttributes);

	let { icon, active = false, children, ...attrs }: Props = $props();
	const navigationDrawerContext = getNavigationDrawerContext();

	const handleAnchorClick = (
		event: MouseEvent & { currentTarget: EventTarget & HTMLAnchorElement },
	) => {
		if (isAnchorAttrs(attrs) && attrs.href) {
			navigationDrawerContext.showNavigationDrawer = false;
			attrs.onclick?.(event);
		}
	};
	const baseClasses =
		'relative typo-label-large text-on-surface-variant shape-full flex w-full items-center gap-3 py-3 pl-4 pr-6';
	let classes = $derived(concatClasses(attrs.class, baseClasses));
</script>

{#snippet inside()}
	<StateLayer class={concatClasses('bg-on-surface shape-full', active && ' opacity-12')} />
	{#if icon}
		<JetiIcon>{icon}</JetiIcon>
	{:else}
		<span class="inline-flex h-[24px] w-[24px] items-center justify-center">
			<span class="h-[12px] w-[12px] rounded-full bg-outline-variant"></span>
		</span>
	{/if}
	{@render children?.()}
{/snippet}

{#if isAnchorAttrs(attrs)}
	<a class={classes} role="link" tabindex="0" {...attrs} onclick={handleAnchorClick}>
		{@render inside()}
	</a>
{:else if isButtonAttrs(attrs)}
	<button class={classes} {...attrs}>
		{@render inside()}
	</button>
{/if}
