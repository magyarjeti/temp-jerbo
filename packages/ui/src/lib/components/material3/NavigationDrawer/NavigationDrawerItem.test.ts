import { render } from '@testing-library/svelte';
import { expect, test } from 'vitest';
import NavigationDrawerItem from './NavigationDrawerItem.svelte';

test('NavigationDrawerItem snapshot', () => {
	const { container } = render(NavigationDrawerItem);
	expect(container).toMatchSnapshot();
});
