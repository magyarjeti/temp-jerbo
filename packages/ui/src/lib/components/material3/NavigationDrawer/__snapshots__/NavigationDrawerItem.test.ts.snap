// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavigationDrawerItem snapshot 1`] = `
<div>
  <button
    class="typo-label-large text-on-surface-variant shape-full state-layers state-layer-hover-on-surface state-layer-focus-on-surface flex w-full items-center gap-3 py-3 pl-4 pr-6"
  >
    <span
      class="inline-flex h-[24px] w-[24px] items-center justify-center"
    >
      <span
        class="bg-outline-variant h-[12px] w-[12px] rounded-full"
      />
    </span>
     
  </button>
  
</div>
`;
